tracer/src/Datadog.Trace/AppSec/SecuritySettings.cs

```diff
@@ -153,31 +154,38 @@ public SecuritySettings(IConfigurationSource? source, IConfigurationTelemetry te
                                       .AsBool(true);
 
             MaxStackTraces = config
-                                  .WithKeys(ConfigurationKeys.AppSec.MaxStackTraces)
-                                  .AsInt32(defaultValue: 2, validator: val => val >= 1)
-                                  .Value;
+                            .WithKeys(ConfigurationKeys.AppSec.MaxStackTraces)
+                            .AsInt32(defaultValue: 2, validator: val => val >= 1)
+                            .Value;
 
             MaxStackTraceDepth = config
-                                  .WithKeys(ConfigurationKeys.AppSec.MaxStackTraceDepth)
-                                  .AsInt32(defaultValue: 32, validator: val => val >= 1)
-                                  .Value;
+                                .WithKeys(ConfigurationKeys.AppSec.MaxStackTraceDepth)
+                                .AsInt32(defaultValue: 32, validator: val => val >= 1)
+                                .Value;
 
             MaxStackTraceDepthTopPercent = config
-                                  .WithKeys(ConfigurationKeys.AppSec.MaxStackTraceDepthTopPercent)
-                                  .AsInt32(defaultValue: 75, validator: val => val >= 0 && val <= 100)
-                                  .Value;
+                                          .WithKeys(ConfigurationKeys.AppSec.MaxStackTraceDepthTopPercent)
+                                          .AsInt32(defaultValue: 75, validator: val => val >= 0 && val <= 100)
+                                          .Value;
 
             WafDebugEnabled = config
                              .WithKeys(ConfigurationKeys.AppSec.WafDebugEnabled)
                              .AsBool(defaultValue: false);
 
             ScaEnabled = config
-                             .WithKeys(ConfigurationKeys.AppSec.ScaEnabled)
-                             .AsBool();
+                        .WithKeys(ConfigurationKeys.AppSec.ScaEnabled)
+                        .AsBool();
 
             NoCustomLocalRules = Rules == null;
+
+            DdDotnetTracerHome = config.WithKeys(ConfigurationKeys.DotnetTracerHome).AsString();
+            InternalTraceNativeEnginePath = config.WithKeys(ConfigurationKeys.InternalTraceNativeEnginePath).AsString();
```

Comment:
These are reading from the full config stack, but they shouldn't, they should specifically be read only from env vars I think? 🤔 

```suggestion
            var envConfig = new ConfigurationBuilder(new EnvironmentConfigurationSource(), telemetry);
            DdDotnetTracerHome = envConfig .WithKeys(ConfigurationKeys.DotnetTracerHome).AsString();
            InternalTraceNativeEnginePath = envConfig .WithKeys(ConfigurationKeys.InternalTraceNativeEnginePath).AsString();
```

There are likely a couple of other "non-settings" that should similarly be read with this pattern, but that's a separate issue 😄